RockWeb/Blocks/Groups/GroupDetail.ascx.cs

```diff
@@ -825,6 +825,102 @@ protected void btnCancel_Click( object sender, EventArgs e )
+                    rockContext.SaveChanges();
+
+                    newGroup.LoadAttributes( rockContext );
+                    if ( group.Attributes != null && group.Attributes.Any() )
+                    {
+                        foreach ( var attributeKey in group.Attributes.Select( a => a.Key ) )
+                        {
+                            string value = group.GetAttributeValue( attributeKey );
+                            Guid guidValue = value.AsGuid();
+                            newGroup.SetAttributeValue( attributeKey, value );
+                        }
+                    }
+                    newGroup.SaveAttributeValues( rockContext );
+
+                    /* Take care of Group Member Attributes */
+                    var entityTypeId = EntityTypeCache.Read( typeof( GroupMember ) ).Id;
+                    string qualifierColumn = "GroupId";
+                    string qualifierValue = group.Id.ToString();
+                    // Get the existing attributes for this entity type and qualifier value
+                    var attributes = attributeService.Get( entityTypeId, qualifierColumn, qualifierValue );
+
+                    foreach ( var attribute in attributes )
+                    {
+                        var newAttribute = attribute.Clone( false );
+                        newAttribute.Id = 0;
+                        newAttribute.Guid = Guid.NewGuid();
+                        newAttribute.IsSystem = false;
+
+                        foreach ( var qualifier in attribute.AttributeQualifiers )
+                        {
+                            var newQualifier = qualifier.Clone( false );
+                            newQualifier.Id = 0;
+                            newQualifier.Guid = Guid.NewGuid();
+                            newQualifier.IsSystem = false;
+                            newAttribute.AttributeQualifiers.Add( qualifier );
+                        }
+
+                        Rock.Attribute.Helper.SaveAttributeEdits( newAttribute, entityTypeId, qualifierColumn, newGroup.Id.ToString(), rockContext );
+                    }
+
+                    foreach ( var auth in auths )
+                    {
+                        var newAuth = auth.Clone( false );
+                        newAuth.Id = 0;
+                        newAuth.Guid = Guid.NewGuid();
+                        newAuth.GroupId = newGroup.Id;
+                        newAuth.CreatedByPersonAlias = null;
+                        newAuth.CreatedByPersonAliasId = null;
+                        newAuth.CreatedDateTime = RockDateTime.Now;
+                        newAuth.ModifiedByPersonAlias = null;
+                        newAuth.ModifiedByPersonAliasId = null;
+                        newAuth.ModifiedDateTime = RockDateTime.Now;
+                        authService.Add( newAuth );
+                    }
+                    rockContext.SaveChanges();
+                } );
+                ShowDetail( newGroup.Id );
```

Comment:
Because all of the auth rules are cached by Rock, you should add a call to `Rock.Security.Authorization.Flush()` here after saving changes.